indico/modules/rb_new/client/js/common/bookings/reducers.js

```diff
@@ -0,0 +1,42 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {combineReducers} from 'redux';
+
+import {requestReducer} from 'indico/utils/redux';
+import * as bookingsActions from './actions';
+
+
+export default combineReducers({
+    requests: combineReducers({
+        details: requestReducer(
+            bookingsActions.FETCH_BOOKING_DETAILS_REQUEST,
+            bookingsActions.FETCH_BOOKING_DETAILS_SUCCESS,
+            bookingsActions.FETCH_BOOKING_DETAILS_ERROR
+        )
+    }),
+    details: (state = {}, action) => {
+        switch (action.type) {
+            case bookingsActions.BOOKING_DETAILS_RECEIVED: {
+                const {id, ...data} = action.data;
+                return {...state, [id]: data};
```

Comment:
maybe `camelizeKeys(data)`? thanks to that you will probably avoid renaming object keys down below `booked_for_user: bookedFor`

Reply:
camelizeKeys messes up the keys in occurrences (when we use strings with dates as keys) so I won't use it

Reply:
then I think it might be a good thing to actually fix this method since I don't think we want to camelize something like `2018-05-01`

Reply:
Yep, camelize shouldn't break a string that's doesn't even contain any letters... (or rather it should probably only modify strings consisting solely of `[a-zA-A_]`)

Reply:
Good idea, I'll do so